nym-vpn-desktop/src-tauri/src/main.rs

```diff
@@ -93,7 +93,19 @@ async fn main() -> Result<()> {
         &app_config_store.full_path.display()
     );
 
-    let app_config = app_config_store.read().await?;
+    let app_config = match app_config_store.read().await {
+        Ok(cfg) => cfg,
+        Err(e) => {
+            warn!("failed to read app config: {e}, falling back to default (empty) config");
+            debug!("clearing the config file");
+            app_config_store
+                .clear()
+                .await
+                .inspect_err(|e| error!("failed to clear the config file: {e}"))
+                .ok();
+            AppConfig::default()
```

Comment:
I think this is good in that it guards against the case when the file for whatever reason is corrupt or not readable, and we can then reset it.

But for the future, it doesn't handle e.g when the config file is readonly due to for example network share issues. 

Basically the correct and user friendly way to handle this whole section of code is to remove all return or error (`?`) that we can't recover from, instead pass them to the gui constructor further down and display it once the UI is up and running. We shouldn't assume that the user is launching this from a terminal